packages/destination-actions/src/destinations/insider/index.ts

```diff
@@ -0,0 +1,46 @@
+import type { DestinationDefinition } from '@segment/actions-core'
+import type { Settings } from './generated-types'
+import { API_BASE, UPSERT_ENDPOINT } from './insider-helpers'
+
+const destination: DestinationDefinition<Settings> = {
+  name: 'Insider',
+  slug: 'actions-insider',
+  mode: 'cloud',
+
+  authentication: {
+    scheme: 'custom',
+    fields: {
+      account_name: {
+        label: 'Insider Account Name',
+        description:
+          'Insider Account Name is your account name which you can find under Account Preferences at InOne Settings.',
+        type: 'string',
+        required: true
+      },
+      ucd_key: {
+        label: 'UCD Upsert Key',
+        description:
+          'UCD Upsert Key is your unique Upsert Key which you can find under Account Preferences at InOne Settings.',
+        type: 'password',
+        required: true
+      }
+    },
+    testAuthentication: async (request) => {
+      const data = await request(`${API_BASE}${UPSERT_ENDPOINT}`)
+      if (data.status === 200) {
```

Comment:
Dont we have just to check token is expired ? Can be better to use. Also upsert user request can be failed because of something else